### PR TITLE
Renovate timer services

### LINKED ISSUE #3845: KERNEL PANIC from bad arg to localTimerService from REPL
## Describe the bug

From the REPL, if I provide a `number` where the `localTimerService` is expecting a `bigint`, the kernel panics... presumably because the timer device throws a `TypeError` and the kernel considers this fatal.

## To Reproduce

1. Start a REPL
2. Enter `E(home.localTimerService).setWakeup(3, handler)`
3. kerflewey: `2021-09-17T18:50:36.864Z SwingSet: kernel: ##### KERNEL PANIC: error during syscall/device.invoke: TypeError: (a number) must be a bigint #####`

## Expected behavior

a user-level error, i.e. a rejected promise in the REPL

## Platform Environment
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`) agorictest-17-75-ge34f7c30e

cc @FUDCo @Chris-Hibbert

### LINKED ISSUE #3847: TimerService.setWakeup() treats baseTime as a RelativeTime; it should be Timestamp
## Describe the bug

`setWakeup()` in the chain and local timerService treat the first parameter (`baseTime`) as an offset from _now_, rather than as an absolute `Timestamp` as documented.

The `manualTimer` treats it as absolute, so any tested code has that expectation. I think this should be treated as a bug in the timer vat code rather than a mistake in the documentation.

How many places in the system have been working from the as-implemented version and need to be updated?

#3845 is related in that it provoked this realization, but it's actually about the kernel's inappropriate response to times expressed as `numbers` rather than `bigint`.

## To Reproduce

Follow the steps [in the docs](https://agoric.com/documentation/repl/timerServices.html#e-home-chain-or-local-timerservice-setwakeup-basetime-handler) and notice that the baseTime arg has to be a small integer in order to see a response. As documented, it should be slightly larger than the value returned from `E(home.chainTimerService).getCurrentTimestamp()`.

## Expected behavior

timerService should treat the time argument to `setWakeup()` as an absolute `Timestamp`.

### PR DESCRIPTION
**Pull request by michaelfig**
In this PR:
- Fixes #3847 , Fixes #3845
- Delete or update obsolete timer documentation.
- Adds a `TimerService.delay(RelativeTime)` to resolve a Promise after a delay
- Remove deprecated `TimerService.createRepeater`
- Remove duplicated `swingset-runner` `manualTimer.js`
- Update the sim-chain's `chainTimerService` correctly
- Update the solo's `localTimerService` correctly and use millisecond resolution

Still needs:
- tests
- documentation repo updates to reflect the changes

### CODE REVIEW

File: packages/SwingSet/src/vats/vat-timerWrapper.js [hunk 1]
```diff
@@ -46,22 +53,65 @@ export function buildRootObject(vatPowers) {
...
+        // We use an update producer to ensure we don't have timers pending that
+        // nobody's waiting for.
+        const { notifier, updater, setUpdateProducer } = makeNotifierKit();
+
+        const notifierDelayHandler = Far('notifierDelayHandler', {
           wake: updater.updateState,
         });
-        D(timerNode).schedule(index, updateHandler);
+
+        // The latest stamp we have produced an notifier update for.
```
Comment (erights) [inline packages/SwingSet/src/vats/vat-timerWrapper.js]:
```suggestion
        // The latest stamp we have produced a notifier update for.
```

**Review by erights — COMMENTED** [review]
I don't understand the changes to notifier. Let's zoom sometime.

Be aware of https://github.com/Agoric/agoric-sdk/blob/master/packages/notifier/README.md

I haven't looked at it in a while. If we do go ahead with this change, please check if that doc needs to be updated.

**Comment by michaelfig** [comment]
> I don't understand the changes to notifier.

In the interest of decoupling this optimisation from the more urgent changes needed to solve the issues at hand, I've moved the notifier changes and the reliance on them to a new PR #3854.

**Comment by erights** [comment]
> > I don't understand the changes to notifier.
> 
> In the interest of decoupling this optimisation from the more urgent changes needed to solve the issues at hand, I've moved the notifier changes and the reliance on them to a new PR #3854.

Excellent, thanks!

**Comment by erights** [comment]
With that gone, would you mind if I unlisted myself as a reviewer of this PR?

File: packages/SwingSet/docs/timer.md [hunk 2]
```diff
@@ -33,29 +33,31 @@ and creating the service. The vat is responsible for creating repeater objects
...
+it accessible to the user in `home.localTimerService` (whose timestamps are
+measured in milliseconds).
 
 ```
 const timerService = E(vats.timerWrapper).createTimerService(devices.timer);
 ```
 
-Then users in the REPL can use the timerService to schedule wakeups.
+Then users in the REPL can use the `localTimerService` or `chainTimerService`
+(both in seconds) to schedule wakeups.
```
Comment (Chris-Hibbert) [inline packages/SwingSet/docs/timer.md]: This seems inconsistent with the above. Is `localTimerService` in seconds or milliseconds?

**Comment by michaelfig** [comment]
> With that gone, would you mind if I unlisted myself as a reviewer of this PR?

Go right ahead!  (I couldn't figure out how to do that, but I tried.)

File: packages/SwingSet/docs/timer.md [hunk 3]
```diff
@@ -33,29 +33,31 @@ and creating the service. The vat is responsible for creating repeater objects
 to wrap the device node as a capability that can be accessed by regular vat
 code.
 
-In `bootstrap()` for a particular SwingSet, we create a timerService, and
-make it accessible to the user in `home`
+In `bootstrap()` for a particular SwingSet, we create a timerService, and make
```
Comment (Chris-Hibbert) [inline packages/SwingSet/docs/timer.md:36]: Please add a few lines about the new `delay(delay)` method.

File: packages/SwingSet/src/vats/vat-timerWrapper.js [hunk 4]
```diff
@@ -13,26 +16,30 @@ export function buildRootObject(vatPowers) {
...
         return D(timerNode).removeWakeup(handler);
       },
       // deprecated in favor of makeRepeater().
       // TODO(#2164): remove before Beta
-      createRepeater(delaySecs, interval) {
-        return timerService.makeRepeater(delaySecs, interval);
+      createRepeater(delay, interval) {
+        delay = Nat(delay);
+        interval = Nat(interval);
+        return timerService.makeRepeater(delay, interval);
```
Comment (Chris-Hibbert) [inline packages/SwingSet/src/vats/vat-timerWrapper.js]: I think this is a good opportunity to get rid of the deprecated `createRepeater()`.  It's only used in one test in agoric-sdk. I doubt it's used outside this repo.

File: packages/zoe/tools/manualTimer.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ import { Far } from '@agoric/marshal';
 import './types.js';
 import './internal-types.js';
 import { makeNotifierKit } from '@agoric/notifier';
+import { makePromiseKit } from '@agoric/promise-kit';
```
Comment (Chris-Hibbert) [inline packages/zoe/tools/manualTimer.js:12]: Can we drop `packages/swingset-runner/demo/zoeTests/manualTimer.js` and import this code instead?

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
Thanks for taking care of this!

File: packages/SwingSet/src/vats/vat-timerWrapper.js (hunk 1 above)
Comment (michaelfig) [reply packages/SwingSet/src/vats/vat-timerWrapper.js]: Done, and moved to #3854.

File: packages/SwingSet/docs/timer.md (hunk 2 above)
Comment (michaelfig) [reply packages/SwingSet/docs/timer.md]: Used to be seconds (but was incorrect).  Now I've moved to milliseconds and corrected the polling to be accurate.

**Review by Chris-Hibbert — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**